akka-stream-tests/src/test/scala/akka/stream/io/compression/CompressionTestingTools.scala

```diff
@@ -0,0 +1,43 @@
+package akka.stream.io.compression
+
+import akka.stream.Materializer
+import akka.stream.scaladsl.Source
+import akka.util.ByteString
+
+import scala.annotation.tailrec
+import scala.concurrent.{ Await, ExecutionContext, Future }
+import scala.concurrent.duration.Duration
+import scala.util.{ Failure, Success }
+
+// a few useful helpers copied over from akka-http
+object CompressionTestingTools {
+  implicit class AddFutureAwaitResult[T](val future: Future[T]) extends AnyVal {
+    /** "Safe" Await.result that doesn't throw away half of the stacktrace */
+    def awaitResult(atMost: Duration): T = {
```

Comment:
does `futureValue` from ScalaTest also throw away half of the stacktrace? (we have integration with it in AkkaSpec so that it uses the default akka test timeout)